pommerman/network/ionc/__init__.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python
+"""IonClient Entry point"""
+
+# Global Imports
+import ui
+from .constants import LangEN as Strings
+from .network import Network as net
+from signal import signal, SIGINT
+from gym.spaces import Discrete
+
+
+# Functions
+def init():
+    """Initialize the application"""
+
+    def exit_handler(_s, _h):
+        """Handle exiting the application"""
+        ui.fatal(ui.yellow, "Exiting..")
+
+    signal(SIGINT, exit_handler)
+    if ui.ask_yes_no(Strings.server_prompt.value):
+        domain = ui.ask_string(Strings.server_domain.value)
+        if domain is None:
+            ui.fatal(Strings.e_invalid_domain.value)
+    else:
+        domain = "play.pommerman.com:5050"
+    ui.info(Strings.server_connecting_p1.value, ui.yellow,
+            Strings.server_connecting_p2.value, ui.reset,
+            Strings.server_connecting_p3.value)
+    network = net(domain)
+    status = network.check()
+    if not status[0]:
+        ui.fatal(Strings[status[1]].value)
+    else:
+        ui.info(Strings.server_connected.value, ui.yellow,
+                Strings.server_players.value,
+                str(status[1]) + ",", Strings.server_matches.value, status[2])
+    intent(network)
+
+
+def intent(network):
+    """Get the intent of the user"""
+    i = ui.ask_choice(Strings.intent.value, [
+        Strings.intent_match.value, Strings.intent_room.value,
+        Strings.intent_replay.value, Strings.intent_exit.value
+    ])
+    if i == Strings.intent_match.value:
+        match(network)
+    elif i == Strings.intent_room.value:
+        match(network, True)
+    elif i == Strings.intent_replay.value:
+        replay(network)
+    elif i == Strings.intent_exit.value:
+        exit(0)
+
+
+def match(network, room=False):
+    """Facilitates playing of matches"""
+    from os import getcwd
+    from sys import path
+    from importlib import import_module
+    if room:
+        room = str(ui.ask_string(Strings.room_code.value))
+    path.append(getcwd())
+    agent = import_module(ui.ask_string(Strings.match_import.value))
+    agent_class = ui.ask_string(Strings.match_class_name.value)
+    if agent_class not in agent.__dir__():
+        ui.fatal("Cannot find class in module file")
+    agent = getattr(agent, agent_class)
+    if getattr(agent, "act"):
+        ui.info(ui.green, Strings.match_agent_success.value)
+    agent = agent()
+    ui.info(ui.yellow, Strings.server_comm.value)
+    network.join_list(room)
+    ui.info(Strings.match_variant.value, ui.yellow, network.mode)
+    ui.info(ui.yellow, Strings.match_wait.value)
+    network.waitm()
+    ui.info(Strings.match_run.value, "#" + network.match_id)
```

Comment:
There are a lot of these ui calls. What are they trying to accomplish?

Reply:
Importing the agent is a CLI heavy part of the application as they need to request the user for information such as the name of the module to be imported and such. It'll make sense if you use the application.